android-libs/src/main/cpp/reporter.cpp

```diff
@@ -0,0 +1,111 @@
+// Part of measurement-kit <https://measurement-kit.github.io/>.
+// Measurement-kit is free software. See AUTHORS and LICENSE for more
+// information on the copying conditions.
+
+#include "io_ooni_mk_MKReporterResults.h"
+#include "io_ooni_mk_MKReporterTask.h"
+
+#include <memory>
+#include <string>
+#include <vector>
+
+#include "vendor/json.hpp"
+#include "vendor/mkcollector.hpp"
+
+#include "mkall_util.h"
+
+JNIEXPORT jlong JNICALL Java_io_ooni_mk_MKReporterTask_New(
+    JNIEnv *env, jclass /*clazz*/, jstring softwareName, jstring softwareVersion) {
+  if (env == nullptr || softwareName == nullptr || softwareVersion == nullptr) {
+    MKALL_THROW_EINVAL;
+    return 0L;
+  }
+  std::string software_name;
+  std::string software_version;
```

Comment:
Using variables that differ only because they're snake case rather than camel case is really bad.